Sources/Common/Communication/Event.swift

```diff
@@ -0,0 +1,155 @@
+import Foundation
+
+/// Protocol defining the structure and requirements of events in the system.
+///
+/// Conformance to this protocol ensures compatibility with the EventBus system.
+/// Events can carry additional data through a `params` dictionary, offering flexibility for different use cases.
+public protocol EventRepresentable: Equatable, Codable {
+    /// A unique key representing the event type.
+    /// This is used to differentiate and manage different kinds of events.
+    var key: String { get }
+    /// A unique identifier for each event instance.
+    /// This identifier aids in storage and retrieval operations.
+    var storageId: String { get }
+    /// A dictionary containing parameters or data associated with the event.
+    var params: [String: String] { get }
+}
+
+// Default implementation for `EventRepresentable`
+public extension EventRepresentable {
+    // Provides a default implementation of the `key` property.
+    /// It uses the type's name as a unique key.
+    static var key: String {
+        String(describing: Self.self)
+    }
+
+    /// Returns the default `key` for the instance.
+    var key: String {
+        Self.key
+    }
+}
+
+// MARK: - Event Types Registry
+
+/// A registry for all event types used within the system.
+/// This enum can be extended to include new event types as they are defined.
+public enum EventTypesRegistry {
+    /// Returns an array of all event types available in the system.
+    /// This is used to manage and interact with different types of events.
+    static func allEventTypes() -> [any EventRepresentable.Type] {
+        // Add all your event types here
+        [
+            ProfileIdentifiedEvent.self,
+            ScreenViewedEvent.self,
+            ResetEvent.self,
+            TrackMetricEvent.self,
+            TrackInAppMetricEvent.self,
+            RegisterDeviceTokenEvent.self,
+            DeleteDeviceTokenEvent.self,
+            NewSubscriptionEvent.self
+        ]
+    }
+}
```

Comment:
Good callout, while we are not using enum to define a list of enumerated cases, using it as a namespace for static methods and properties related to event types to improve readability and maintainability and the future use cases this registery could have offered but since its internal, i agree we should probably make it a constant and if needed make it an enum.

Reply:
SwiftLint automatically suggests/converts a struct with static members to an enum, hence the enum.